File: my-folder/0125-valid-palindrome/solution.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string filtered_str, reversed_str;
        
        for (auto ch:s){
            if (isalnum(ch)) filtered_str+=tolower(ch);
        }

        reversed_str.resize(filtered_str.size());
        reverse_copy(filtered_str.begin(), filtered_str.end(), reversed_str.begin());
        return reversed_str == filtered_str;
    }
};

File: my-folder/0125-valid-palindrome/solution.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        n = len(s)
        cleaned_str = []

        for i in range(n):
            if s[i].isalnum():
                cleaned_str.append(s[i].lower())
        
        start_ptr = 0
        end_ptr = len(cleaned_str) - 1

        while start_ptr < end_ptr:
            if cleaned_str[start_ptr] != cleaned_str[end_ptr]:
                return False
            
            start_ptr+=1
            end_ptr-=1
        
        return True
